src/store/modules/TsModsModule.ts

```diff
@@ -148,16 +145,95 @@ export const TsModsModule = {
         setExclusions(state, payload: string[]) {
             state.exclusions = payload;
         },
-        startBackgroundUpdate(state) {
-            state.isBackgroundUpdateInProgress = true;
+        setThunderstoreModListUpdateError(state, error: string|unknown) {
+            if (typeof error === 'string') {
+                state.thunderstoreModListUpdateError = error;
+            } else {
+                const msg = error instanceof Error ? error.message : "Unknown error";
+                state.thunderstoreModListUpdateError = msg;
+            }
+        },
+        setThunderstoreModListUpdateStatus(state, status: string) {
+            state.thunderstoreModListUpdateStatus = status;
+        },
+        startThunderstoreModListUpdate(state) {
+            state.isThunderstoreModListUpdateInProgress = true;
+            state.thunderstoreModListUpdateError = '';
         },
         updateDeprecated(state, allMods: ThunderstoreMod[]) {
             state.deprecated = Deprecations.getDeprecatedPackageMap(allMods);
         }
     },
 
     actions: <ActionTree<State, RootState>>{
-        async fetchPackageListIndex({dispatch, rootState}): Promise<PackageListIndex> {
+        /**
+         * Complete update process of the mod list, to be used after
+         * passing the splash screen.
+         */
+        async fetchAndProcessPackageList({commit, dispatch, state}): Promise<void> {
+            if (state.isThunderstoreModListUpdateInProgress) {
+                return;
+            }
+
+            commit('startThunderstoreModListUpdate');
+
+            try {
+                commit('setThunderstoreModListUpdateStatus', 'Checking for mod list updates from Thunderstore...');
+                let packageListIndex: PackageListIndex;
+
+                try {
+                    packageListIndex = await dispatch('fetchPackageListIndex');
+                } catch {
+                    // "Retry failed after 5 attempts" isn't very helpful error message.
```

Comment:
Noted and agreed, worth removing the comment now as it doesn't benefit the codebase

Reply:
Comment removed.